apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -628,4 +648,12 @@ if (
 			return null;
 		},
 	} );
+
+	registerDelegateEventSubscriber(
+		'wpcom-block-editor-template-part-detach-blocks',
+		'before',
+		() => {
+			ignoreNextReplaceBlocksAction = true;
+		}
+	);
```

Comment:
It seems this is bleeding into other events using the same selector, as we did not check the conditions here as we did for firing the event.

https://github.com/Automattic/wp-calypso/pull/54415 should fix